tx/textual/valuerenderer/coins_test.go

```diff
@@ -45,14 +47,52 @@ func TestCoinsJsonTestcases(t *testing.T) {
 				require.NoError(t, err)
 				require.Equal(t, 1, len(screens))
 				require.Equal(t, tc.Text, screens[0].Text)
-			}
 
-			// TODO Add parsing tests
-			// https://github.com/cosmos/cosmos-sdk/issues/13153
+				for _, v := range tc.Metadata {
+					ctx = context.WithValue(ctx, mockCoinMetadataKey(v.Display), v)
+				}
+
+				value, err := vr.Parse(ctx, screens)
+				if tc.Error {
+					require.Error(t, err)
+					return
+				}
+
+				require.NoError(t, err)
+				checkCoinsEqual(t, listValue, value.List())
+			}
 		})
 	}
 }
 
+func checkCoinsEqual(t *testing.T, l1, l2 protoreflect.List) {
```

Comment:
```suggestion
// checkCoinsEqual checks that the 2 lists of Coins contain the same
// **set** of coins. It does not check that the order of coins are
// equal, because in Textual, we sort the coins alphabetically after
// rendering, so we lose initial Coins ordering. Instead, we just check
// set equality using a map.
func checkCoinsEqual(t *testing.T, l1, l2 protoreflect.List) {
```